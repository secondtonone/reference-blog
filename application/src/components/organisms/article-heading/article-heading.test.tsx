import renderer from 'react-test-renderer';
import ArticleHeading from '.';
import fixtureAuthors from '~/__fixtures__/authors';

const contentWithText = {
  id: '1',
  title: 'The Ultimate Guide to Google Search Operators and Google Search Commands',
  preview: 'In this guide, weâ€™ll delve deep into the basics of search operators and search commands, helping you understand how to wield these powerful tools effectively and move on to advanced commands and operators.',
  category: {
    name: 'Advanced SEO',
    slug: 'seo/advanced-seo',
    lang: 'en'
  },
  isHot: true,
  timeToRead: 5,
  previewImage: '',
  altPreviewImage: '',
  author: fixtureAuthors[0],
  url: 'article',
  publishedAt: 1597908972,
  share: 0
};

const contentWithImage = {
  id: '1',
  title: 'The Ultimate Guide to Google Search Operators and Google Search Commands',
  preview: 'In this guide, weâ€™ll delve deep into the basics of search operators and search commands, helping you understand how to wield these powerful tools effectively and move on to advanced commands and operators.',
  category: {
    name: 'Advanced SEO',
    slug: 'seo/advanced-seo',
    lang: 'en'
  },
  isHot: true,
  timeToRead: 5,
  previewImage: '/test-image.jpg',
  altPreviewImage: 'altPreviewImage',
  author: fixtureAuthors[0],
  url: 'article',
  publishedAt: 1597908972,
  share: 0
};

describe('Match snapshots ArticleHeading', () => {
  it('with text', () => {
    const tree = renderer.create(
      <ArticleHeading
        {...contentWithText}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with image', () => {
    const tree = renderer.create(
      <ArticleHeading
        {...contentWithImage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with text dark', () => {
    const tree = renderer.create(
      <ArticleHeading
        {...contentWithText}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with image dark', () => {
    const tree = renderer.create(
      <ArticleHeading
        {...contentWithImage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with category', () => {
    const tree = renderer.create(
      <ArticleHeading
        {...{ ...contentWithText, withCategory: true }}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with hidden author profile', () => {
    const tree = renderer.create(
      <ArticleHeading
        {...{
          ...contentWithText,
          author: fixtureAuthors[2]
        }}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
