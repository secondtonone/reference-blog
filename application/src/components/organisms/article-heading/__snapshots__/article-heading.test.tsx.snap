// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots ArticleHeading with category 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gyhqJS article-heading__ArticleHeadingStyled-sc-tmj7yn-2 jeWvxi"
  data-test="article-heading"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 bcjrbf article-heading__TextContainer-sc-tmj7yn-0 fanvyV"
      >
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
          data-test="breadcrumbs"
        >
          <ul
            className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
            data-test="breadcrumbs-links"
          >
            <li
              data-test="breadcrumbs-link-level-1"
            >
              <a
                href="/blog/category/seo/advanced-seo/"
              >
                Advanced SEO
              </a>
            </li>
          </ul>
        </div>
        <h1
          className="text-content__TextStyled-sc-1stkrsz-0 eMookp article-heading__TitleStyled-sc-tmj7yn-5 eucFWa"
          data-test="heading-text"
        >
          The Ultimate Guide to Google Search Operators and Google Search Commands
        </h1>
        <span
          className="text-content__TextStyled-sc-1stkrsz-0 vNtOM article-heading__MetaStyled-sc-tmj7yn-4 jUGrdQ"
          data-test="text-content"
        >
          <strong
            className="article-heading__CategoryStyled-sc-tmj7yn-3 iMvGvt"
            data-test="category"
          >
            <a
              data-test="category-link"
              href="/blog/category/seo/advanced-seo"
              onClick={[Function]}
              onMouseEnter={[Function]}
              rel="noopener"
              target="_blank"
            >
              Advanced SEO
            </a>
          </strong>
          <p
            data-test="author"
          >
            <a
              href="/user/olga-andrienko"
              onClick={[Function]}
              onMouseEnter={[Function]}
              rel="noopener"
              target="_blank"
            >
              Olga Andrienko
            </a>
          </p>
          <span
            data-test="date"
          >
            Aug 20, 2020
          </span>
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </span>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gWsxPs article-heading__ImageStyled-sc-tmj7yn-1 kEMusM"
      >
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Match snapshots ArticleHeading with hidden author profile 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gyhqJS article-heading__ArticleHeadingStyled-sc-tmj7yn-2 jeWvxi"
  data-test="article-heading"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 bcjrbf article-heading__TextContainer-sc-tmj7yn-0 fanvyV"
      >
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
          data-test="breadcrumbs"
        >
          <ul
            className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
            data-test="breadcrumbs-links"
          >
            <li
              data-test="breadcrumbs-link-level-1"
            >
              <a
                href="/blog/category/seo/advanced-seo/"
              >
                Advanced SEO
              </a>
            </li>
          </ul>
        </div>
        <h1
          className="text-content__TextStyled-sc-1stkrsz-0 eMookp article-heading__TitleStyled-sc-tmj7yn-5 eucFWa"
          data-test="heading-text"
        >
          The Ultimate Guide to Google Search Operators and Google Search Commands
        </h1>
        <span
          className="text-content__TextStyled-sc-1stkrsz-0 vNtOM article-heading__MetaStyled-sc-tmj7yn-4 jUGrdQ"
          data-test="text-content"
        >
          <p
            data-test="author"
          >
            Mr. Hidden
          </p>
          <span
            data-test="date"
          >
            Aug 20, 2020
          </span>
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </span>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gWsxPs article-heading__ImageStyled-sc-tmj7yn-1 kEMusM"
      >
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Match snapshots ArticleHeading with image 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gyhqJS article-heading__ArticleHeadingStyled-sc-tmj7yn-2 jeWvxi"
  data-test="article-heading"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 bcjrbf article-heading__TextContainer-sc-tmj7yn-0 fanvyV"
      >
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
          data-test="breadcrumbs"
        >
          <ul
            className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
            data-test="breadcrumbs-links"
          >
            <li
              data-test="breadcrumbs-link-level-1"
            >
              <a
                href="/blog/category/seo/advanced-seo/"
              >
                Advanced SEO
              </a>
            </li>
          </ul>
        </div>
        <h1
          className="text-content__TextStyled-sc-1stkrsz-0 eMookp article-heading__TitleStyled-sc-tmj7yn-5 eucFWa"
          data-test="heading-text"
        >
          The Ultimate Guide to Google Search Operators and Google Search Commands
        </h1>
        <span
          className="text-content__TextStyled-sc-1stkrsz-0 vNtOM article-heading__MetaStyled-sc-tmj7yn-4 jUGrdQ"
          data-test="text-content"
        >
          <p
            data-test="author"
          >
            <a
              href="/user/olga-andrienko"
              onClick={[Function]}
              onMouseEnter={[Function]}
              rel="noopener"
              target="_blank"
            >
              Olga Andrienko
            </a>
          </p>
          <span
            data-test="date"
          >
            Aug 20, 2020
          </span>
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </span>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gWsxPs article-heading__ImageStyled-sc-tmj7yn-1 kEMusM"
      >
        <img
          alt="altPreviewImage"
          className="image__ImageStyled-sc-rj8okk-0 beGhfT"
          data-src={null}
          data-test="image"
          height="248"
          src="/test-image.jpg"
          width="340"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Match snapshots ArticleHeading with image dark 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gyhqJS article-heading__ArticleHeadingStyled-sc-tmj7yn-2 jeWvxi"
  data-test="article-heading"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 bcjrbf article-heading__TextContainer-sc-tmj7yn-0 fanvyV"
      >
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
          data-test="breadcrumbs"
        >
          <ul
            className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
            data-test="breadcrumbs-links"
          >
            <li
              data-test="breadcrumbs-link-level-1"
            >
              <a
                href="/blog/category/seo/advanced-seo/"
              >
                Advanced SEO
              </a>
            </li>
          </ul>
        </div>
        <h1
          className="text-content__TextStyled-sc-1stkrsz-0 eMookp article-heading__TitleStyled-sc-tmj7yn-5 eucFWa"
          data-test="heading-text"
        >
          The Ultimate Guide to Google Search Operators and Google Search Commands
        </h1>
        <span
          className="text-content__TextStyled-sc-1stkrsz-0 vNtOM article-heading__MetaStyled-sc-tmj7yn-4 jUGrdQ"
          data-test="text-content"
        >
          <p
            data-test="author"
          >
            <a
              href="/user/olga-andrienko"
              onClick={[Function]}
              onMouseEnter={[Function]}
              rel="noopener"
              target="_blank"
            >
              Olga Andrienko
            </a>
          </p>
          <span
            data-test="date"
          >
            Aug 20, 2020
          </span>
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </span>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gWsxPs article-heading__ImageStyled-sc-tmj7yn-1 kEMusM"
      >
        <img
          alt="altPreviewImage"
          className="image__ImageStyled-sc-rj8okk-0 beGhfT"
          data-src={null}
          data-test="image"
          height="248"
          src="/test-image.jpg"
          width="340"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Match snapshots ArticleHeading with text 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gyhqJS article-heading__ArticleHeadingStyled-sc-tmj7yn-2 jeWvxi"
  data-test="article-heading"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 bcjrbf article-heading__TextContainer-sc-tmj7yn-0 fanvyV"
      >
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
          data-test="breadcrumbs"
        >
          <ul
            className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
            data-test="breadcrumbs-links"
          >
            <li
              data-test="breadcrumbs-link-level-1"
            >
              <a
                href="/blog/category/seo/advanced-seo/"
              >
                Advanced SEO
              </a>
            </li>
          </ul>
        </div>
        <h1
          className="text-content__TextStyled-sc-1stkrsz-0 eMookp article-heading__TitleStyled-sc-tmj7yn-5 eucFWa"
          data-test="heading-text"
        >
          The Ultimate Guide to Google Search Operators and Google Search Commands
        </h1>
        <span
          className="text-content__TextStyled-sc-1stkrsz-0 vNtOM article-heading__MetaStyled-sc-tmj7yn-4 jUGrdQ"
          data-test="text-content"
        >
          <p
            data-test="author"
          >
            <a
              href="/user/olga-andrienko"
              onClick={[Function]}
              onMouseEnter={[Function]}
              rel="noopener"
              target="_blank"
            >
              Olga Andrienko
            </a>
          </p>
          <span
            data-test="date"
          >
            Aug 20, 2020
          </span>
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </span>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gWsxPs article-heading__ImageStyled-sc-tmj7yn-1 kEMusM"
      >
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Match snapshots ArticleHeading with text dark 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gyhqJS article-heading__ArticleHeadingStyled-sc-tmj7yn-2 jeWvxi"
  data-test="article-heading"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 djPBEv"
    data-test="container"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 bcjrbf article-heading__TextContainer-sc-tmj7yn-0 fanvyV"
      >
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
          data-test="breadcrumbs"
        >
          <ul
            className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
            data-test="breadcrumbs-links"
          >
            <li
              data-test="breadcrumbs-link-level-1"
            >
              <a
                href="/blog/category/seo/advanced-seo/"
              >
                Advanced SEO
              </a>
            </li>
          </ul>
        </div>
        <h1
          className="text-content__TextStyled-sc-1stkrsz-0 eMookp article-heading__TitleStyled-sc-tmj7yn-5 eucFWa"
          data-test="heading-text"
        >
          The Ultimate Guide to Google Search Operators and Google Search Commands
        </h1>
        <span
          className="text-content__TextStyled-sc-1stkrsz-0 vNtOM article-heading__MetaStyled-sc-tmj7yn-4 jUGrdQ"
          data-test="text-content"
        >
          <p
            data-test="author"
          >
            <a
              href="/user/olga-andrienko"
              onClick={[Function]}
              onMouseEnter={[Function]}
              rel="noopener"
              target="_blank"
            >
              Olga Andrienko
            </a>
          </p>
          <span
            data-test="date"
          >
            Aug 20, 2020
          </span>
          <span
            data-test="time-to-read"
          >
            5
             
            min read
          </span>
        </span>
      </div>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 gWsxPs article-heading__ImageStyled-sc-tmj7yn-1 kEMusM"
      >
        
      </div>
    </div>
  </div>
</div>
`;
